src/StreamrClient.js

```diff
@@ -297,54 +288,49 @@ export default class StreamrClient extends EventEmitter {
-            this.subscribe(ethAddress, async (parsedContent, streamMessage) => {
+            const publisherId = await this.getPublisherId()
+            const streamId = KeyExchangeUtil.getKeyExchangeStreamId(publisherId)
+            this.subscribe(streamId, async (parsedContent, streamMessage) => {
                 try {
                     if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_REQUEST) {
                         if (this.keyExchangeUtil) {
                             await this.keyExchangeUtil.handleGroupKeyRequest(streamMessage)
                         }
                     } else if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_RESPONSE_SIMPLE) {
                         if (this.keyExchangeUtil) {
-                            const { streamId } = streamMessage.getParsedContent()
-                            // A valid publisher of the client's inbox stream could send key responses for other streams to which
-                            // the publisher doesn't have write permissions. Thus the following additional check is necessary.
-                            // TODO: fix this hack in other PR (and move logic to keyExchangeUtil)
-                            const valid = await this.subscribedStreamPartitions[streamId + '0'].isValidPublisher(streamMessage.getPublisherId())
-                            if (valid) {
-                                this.keyExchangeUtil.handleGroupKeyResponse(streamMessage)
-                            } else {
-                                throw new InvalidGroupKeyResponseError(
-                                    `Received group key from an invalid publisher ${streamMessage.getPublisherId()} for stream ${streamId}`
-                                )
-                            }
+                            this.keyExchangeUtil.handleGroupKeyResponse(streamMessage)
                         }
-                    } else if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.ERROR_MSG) {
+                    } else if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_ERROR_RESPONSE) {
                         debug('WARN: Received error of type %s from %s: %s',
                             streamMessage.getParsedContent().code, streamMessage.getPublisherId(), streamMessage.getParsedContent().message)
                     } else {
                         throw new InvalidContentTypeError(`Cannot handle message with content type: ${streamMessage.contentType}`)
                     }
-                } catch (err) {
-                    if (err instanceof InvalidGroupKeyRequestError
-                        || err instanceof InvalidGroupKeyResponseError
-                        || err instanceof InvalidContentTypeError) {
-                        debug('WARN: %s', err.message)
+                } catch (error) {
+                    if (error instanceof InvalidGroupKeyRequestError
```

Comment:
The thing we talked about just now: We should only send an error response to a `GroupKeyRequest`, but not to the other types of messages (which are basically responses).

How I did it in the Java client was add an inner `try-catch` to the if-branch where `GroupKeyRequest` is handled

```
if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_REQUEST) {
    try {
        ....
    } catch (...) {
        // send error response
    }
}
```

but you might choose to do it another way.

Reply:
Done.